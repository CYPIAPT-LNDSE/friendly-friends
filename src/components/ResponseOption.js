import React from 'react';

const ResponseOption = (props) => {
  return (
    <p className="response-option">{props.text}</p>
  );
}

export default ResponseOption;
