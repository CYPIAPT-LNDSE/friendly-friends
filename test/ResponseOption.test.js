import React from 'react';
import renderer from 'react-test-renderer';
import ResponseOption from '../src/components/ResponseOption.js';

test('ResponseOption renders correctly', () => {
  const tree = renderer.create(
    <ResponseOption
      response={{text: 'hello', next: 2}}
    />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Updating ResponseOption works correctly', () => {
  const mockClick = jest.fn();
  const tree = renderer.create(
    <ResponseOption
      response={{text: 'hello', next: 2}}
      updateMessage={mockClick}
    />).toJSON();
  tree.props.onClick();
  expect(tree).toMatchSnapshot();
  expect(mockClick.mock.calls.length).toBe(1);
});
