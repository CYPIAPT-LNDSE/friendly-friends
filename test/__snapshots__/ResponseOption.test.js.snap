exports[`test ResponseOption renders correctly 1`] = `
<p
  className="response-option"
  onClick={[Function]}>
  hello
</p>
`;

exports[`test Updating ResponseOption works correctly 1`] = `
<p
  className="response-option"
  onClick={[Function]}>
  hello
</p>
`;
